### ISSUE TITLE
Pattern Iterators Broken in Origen 0.28+

### ISSUE THREAD

**Issue by coreyeng** [Origen-SDK/origen#185]
Hello. I recently found that pattern iterators (https://github.com/Origen-SDK/origen/blob/0731f00a12a7737331fb85f6659706d9eb398495/lib/origen/generator/pattern_iterator.rb) no longer work as they did in 0.27. By that I mean... at all. The most relevant files don't seem to have changed since 2015 and looking through the release notes I don't see anything that should have affected this (maybe some of the String/Symbol methods?).

In any case, the sub block in question was developed in legacy RGen days and is a bit far behind in some aspects. I didn't see anything about pattern iterators in the documentation or in the specs. Are pattern iterators planned on being supported? Or should we just remove them in favor of calling Pattern.create in a loop and passing it options from the command line?

The TL;DR of this is... pattern iterators are currently broken. Should they be fixed or just removed?

Thanks.

**Comment by coreyeng** [comment]
Another note is that the pattern compiler in Origen Tester also does not take pattern iterators into account. So, for example, the BINL converter dies partway through. However this seems to fail both with pattern iterators or with pattern.create in a loop. It converts the pattern but doesn't move everything together nicely so its less of an issue. Nothing a few cp commands can't fix.

**Comment by ginty** [comment]
There are no plans to remove support for pattern iterators and they will continue to be supported.
I would say that advanced users tend to graduate to [source-less pattern generation](http://origen-sdk.org/origen/guides/pattern/name/) and once you do that there is no need to use pattern iterators any more.

I'm not convinced yet that they are broken however, the test suites of Origen and OrigenTesters contain some tests of pattern iterators and these pass - https://github.com/Origen-SDK/origen/blob/master/list/j750.list

**Comment by info-rchitect** [comment]
@coreyeng are pattern iterators touching the file handler module?  I assume so, so it is possible my change in the compilers fix broke it.  But when did you see this?  If it was before ~3pm CST yesterday then it is not that change.

**Comment by info-rchitect** [comment]
@ginty, just an FYI your source-less pattern generation is how I designed our app model loading.  i made a dictionary of model types and made loading or importing models very easy to do, regardless of where they come from.  sry for the interruption, back to debug...

**Comment by coreyeng** [comment]
@info-rchitect I just noticed this last night (~8:00 PM CST), but we are using version 0.28.0. It would not have been that change.

@ginty I'm pretty sure its Origen just from the following:

Origen 0.27: origen g pattern_trim #=> pattern_trim_m4.avc
Origen 0.28: origen g pattern_trim #=> pattern_trim.avc

This is holding all gems the same and just switching Origen. I suppose it could be the way I'm using the pattern iterators? I didn't see anything that even affected this though between 0.27 and 0.28. I haven't needed to mess with source-less generation either. Our apps generally aren't large enough to need it.

**Comment by coreyeng** [comment]
Fixed in #231
